### PR TITLE
More pre-commit and validate-pyproject

### PR DESCRIPTION
**Pull request by cclauss**
Ensure files remain tidy and `pyproject.toml` is valid.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,9 +1,45 @@
 repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.6.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-ast
+      - id: check-builtin-literals
+      - id: check-byte-order-marker
```
Comment (greenbender) [inline .pre-commit-config.yaml:8]: `check-byte-order-marker` appears to be deprecated

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,9 +1,45 @@
...
+      - id: end-of-file-fixer
+      - id: file-contents-sorter
+      - id: fix-byte-order-marker
+      - id: forbid-new-submodules
+      - id: forbid-submodules
+      - id: mixed-line-ending
+      # - id: pretty-format-json
+      - id: requirements-txt-fixer
+      - id: sort-simple-yaml
+      - id: trailing-whitespace
```
Comment (greenbender) [inline .pre-commit-config.yaml:32]: Adding `no-commit-to-branch` would be useful. I often make this mistake.

**Comment by cclauss** [comment]
A nice weekend of work!  Thanks massively.

**Comment by greenbender** [comment]
Thank you for all your help! Huge improvements to this repo. Expect release 2.0.0 when I get a little time. Next weekend maybe.

**Pull request merged**